coalib/parsing/filters/LanguageFilter.py

```diff
@@ -0,0 +1,20 @@
+from coalib.parsing.filters.FilterBase import FilterBase
+
+
+class LanguageFilter(FilterBase):
+
+    def filter_section_bears(self, bears, args):
+        """
+        Filters the bears by can_detect.
+
+        :param bears:       The dictionary of the sections as keys and list of
+                            bears as values.
+        :param args:        languages that bears are being filtered on.
+        :return:            New dictionary with filtered out bears that matches
+                            any language from languages.
+        """
+        languages = {lang.lower() for lang in args} | {'all'}
+        return {section:
+                tuple(bear for bear in bears[section]
+                      if {lang.lower() for lang in bear.LANGUAGES} & languages)
+                for section in bears.keys()}
```

Comment:
It will be `bears` and not `bears.values()` as I'm going through keys and not values for all three filters.